src/main/java/com/google/cloud/genomics/utils/ShardUtils.java

```diff
@@ -285,13 +294,47 @@ public StreamReadsRequest apply(String readGroupSetId) {
     return contigs;
   }
 
-  private static List<Contig> getAllShardsInVariantSet(String variantSetId,
+  private static List<Contig> getAllShardsInReadGroupSet(String readGroupSetId,
       SexChromosomeFilter sexChromosomeFilter, long numberOfBasesPerShard,
       GenomicsFactory.OfflineAuth auth) throws IOException, GeneralSecurityException {
-    List<Contig> contigs = getContigsInVariantSet(variantSetId, sexChromosomeFilter, auth);
+    List<Contig> contigs = getContigsInReadGroupSet(readGroupSetId, sexChromosomeFilter, auth);
     return ShardUtils.getAllShardsForContigs(contigs, numberOfBasesPerShard);
   }
-
+  
+  /**
+   * Retrieve the list of all the reference names and their start=0/end positions for the referenceSet
+   * associated with the readGroupSet.
+   * 
+   * Note that start is hardcoded to zero since references only specify their length. 
+   * 
+   * @param readGroupSetId - The id of the readGroupSet to query.
+   * @param sexChromosomeFilter - An enum value indicating how sex chromosomes should be
+   *        handled in the result.
+   * @return The list of all references in the readGroupSet.
+   * @throws IOException
+   * @throws GeneralSecurityException 
+   */
+  private static List<Contig> getContigsInReadGroupSet(String readGroupSetId,
+      SexChromosomeFilter sexChromosomeFilter, GenomicsFactory.OfflineAuth auth)
+          throws IOException, GeneralSecurityException {
+    String referenceSetId = GenomicsUtils.getReferenceSetId(readGroupSetId, auth);
+    if(null == referenceSetId) {
+      throw new IllegalArgumentException("No referenceSetId associated with readGroupSetId "
+          + readGroupSetId + ".");
+    }
+    List<Contig> contigs = Lists.newArrayList();
+    for (Reference reference : GenomicsUtils.getReferences(referenceSetId, auth)) {
+      String referenceName = reference.getName().toLowerCase();
+      if (sexChromosomeFilter == SexChromosomeFilter.EXCLUDE_XY
+          && (referenceName.contains("x") || referenceName.contains("y"))) {
```

Comment:
This is brittle. Consider the following hg19 reference:
`chr6_cox_hap2`

https://developers.google.com/apis-explorer/#p/genomics/v1/genomics.references.get?referenceId=EP_a7pq5kZq4Xg&_h=1&

I would recommend trimming a 'chr' prefix, if any, then do an exact match on x or y.